### PR TITLE
bump OpenSplice version and use -o maintain-include-namespace

### PR DESCRIPTION
**Pull request by dirk-thomas**
While #20 targets Crystal in a backward compatible way this patch requires the newer version of OpenSplice.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
This is only feasible for `master` / Dashing since for Crystal it would require macOS / Windows users to update their OpenSplice installation which isn't realistic for an already released distro.

**Comment by dirk-thomas** [comment]
Linux builds using ros2/ci#251:
* Linux [Build Status](https://ci.ros2.org/job/ci_linux/6298/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/2716/)

---

* macOS *needs newer OpenSplice version*
* Windows *needs newer OpenSplice version*

**Comment by sloretz** [comment]
> * macOS _needs newer OpenSplice version_
> * Windows _needs newer OpenSplice version_

mini1/mini2/mini3 and icecube/portable/windshield have all had their opensplice version upgraded to `6.9.190227`

**Comment by dirk-thomas** [comment]
> mini1/mini2/mini3 and icecube/portable/windshield have all had their opensplice version upgraded to `6.9.190227`

The latest OpenSplice release doesn't work yet as expected. See the failing jobs above as well as the conversation on ros2/rmw_opensplice#259. So it is likely that a new version will be required.

**Comment by dirk-thomas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6394/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5261/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6234/)

**Comment by dirk-thomas** [comment]
Moving back to "in progress" until there is an updated OpenSplice release.

**Comment by dirk-thomas** [comment]
Follow up of ros2/rosidl_typesupport_opensplice#23. Instead of using the new option when available this patch requires the latest OpenSplice release and uses the option unconditionally.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6616/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2946/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5432/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6438/)

(Addressed the linter warning about unused `re`.)

Back in review.

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake [hunk 1]
```diff
@@ -177,9 +177,6 @@ target_include_directories(${rosidl_generate_interfaces_TARGET}${_target_suffix}
   PUBLIC
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_cpp
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_opensplice_cpp
-  "${_output_path}/msg/dds_opensplice"
-  "${_output_path}/srv/dds_opensplice"
-  "${_output_path}/action/dds_opensplice"
```
Comment (hidmic) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake:182]: @dirk-thomas unrelated changes or am I missing something?

File: rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake [hunk 2]
```diff
@@ -148,12 +148,6 @@ target_include_directories(${rosidl_generate_interfaces_TARGET}${_target_suffix}
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_cpp
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_opensplice_c
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_opensplice_cpp
-  "${_output_path}/msg/dds_opensplice_c"
-  "${_dds_output_path}/msg/dds_opensplice"
-  "${_output_path}/srv/dds_opensplice_c"
-  "${_dds_output_path}/srv/dds_opensplice_c"
-  "${_output_path}/action/dds_opensplice_c"
-  "${_dds_output_path}/action/dds_opensplice_c"
```
Comment (hidmic) [inline rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake:156]: @dirk-thomas unrelated changes or am I missing something?

**Review by hidmic — APPROVED** [review]
LGTM!

**Comment by hidmic** [comment]
Just in case, mind to re-run CI after the linter fix?

File: rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake (hunk 2 above)
Comment (dirk-thomas) [reply rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake:156]: Not unrelated at all. With the new option being used the generated files use full include paths and therefore these include directories are not needed anymore.

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake (hunk 1 above)
Comment (dirk-thomas) [reply rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake:182]: See same comment below: https://github.com/ros2/rosidl_typesupport_opensplice/pull/21#discussion_r272736688

**Comment by dirk-thomas** [comment]
> mind to re-run CI after the linter fix?

I rather not spend the CI time for a trivial change like that.
